fix NxDropdownMenu btn on hover - RSC 1515

EmilyBermudez88 opened the pull request:
https://issues.sonatype.org/browse/RSC-1515

The text within the dropdown toggle was failing consistently in one snapshot, so updated that here as well as part of fixing the mistake with NxDropdown.

avnvbhatta commented:
Since we've modified the hover state, I was expecting some of the hover based snapshot changes, but I don't see them.
Also the current changed snapshot only has the "Filter" placeholder text changed?

EmilyBermudez88 commented:
> Since we've modified the hover state, I was expecting some of the hover based snapshot changes, but I don't see them. Actually, looking at the test, it looks like we don't have a hover test when the dropdown is open. Perhaps we should add a hover test.

There weren't any tests that deal with the interaction state of the dropdown menu, but that's a good point. I'll add them in.


> Also the current changed snapshot only has the "Filter" placeholder text changed?

Yep that was it. And it was the only snapshot that had this happen (all other text colours of the placeholder passed). I tested it 5 or 6 times and consistently that one failed. It must have had a glitch when I initially ran the tests the first time.

EmilyBermudez88 commented:
> > Since we've modified the hover state, I was expecting some of the hover based snapshot changes, but I don't see them. Actually, looking at the test, it looks like we don't have a hover test when the dropdown is open. Perhaps we should add a hover test.
> 
> There weren't any tests that deal with the interaction state of the dropdown menu, but that's a good point. I'll add them in.
> 

Added here! [5e7f9cf](https://github.com/sonatype/sonatype-react-shared-components/pull/1005/commits/5e7f9cfa211ad602da2329e595cf0b35283c2130)

avnvbhatta reviewed (approved):
LGTM
